packages/core/src/internal/effectRunnerMap.js

```diff
@@ -122,6 +122,11 @@ function runCallEffect(env, { context, fn, args }, cb, { task }) {
 
     cb(result)
   } catch (error) {
+    // Remove again?
+    if (env._eventStackDepth > 1000) {
+      /* eslint-disable no-console */
+      console.warn('More than 1000 effects are currently processed recursively.')
+    }
```

Comment:
Theoretically, one could use the added `_eventStackDepth` property to warn proactively, but it would only be a heuristic (it highly depends on what the callstack looks like exactly and there is no linear correlation between the callstack size and `_eventStackDepth`).
Therefore, I would remove the code related to `_eventStackDepth` again, but I wanted to leave it here as food for thought for now.